measurements/api/private.py

```diff
@@ -44,34 +44,49 @@ def api_private_stats_by_month(orm_stat):
     end_date = datetime(now.year, now.month, 1)
     start_date = end_date - relativedelta(months=24)
 
-    r = current_app.db_session.query(
-            func.date_trunc('month', Report.test_start_time).label('test_start_month'),
-            orm_stat
-        ).filter(Report.test_start_time >= start_date).filter(Report.test_start_time < end_date
-        ).group_by('test_start_month')
+    s = select([
+        func.date_trunc('month', sql.text('bucket_date')).label('bucket_month'),
+        sql.text(orm_stat)
+    ]).where(and_(
+        sql.text("bucket_date > :start_date"),
+        sql.text("bucket_date < :end_date")
+    )).group_by(
+        'bucket_month'
+    ).select_from(sql.table('ooexpl_bucket_msm_count'))
```

Comment:
This seems to me another way to prepare the query. What is the reason? Why is it faster?

Reply:
(I have seen some benchmarks above, but I'm still a bit in the dark.)

Reply:
The query is now using a different table which has some pre-computed stats inside of it. Since the stats in that table are precomputed it's now significantly faster.

Reply:
> What was the table before?

`report`. This is implied by the fact that in the query you are using the SQLAlchemy ORM `Report` object.

Reply:
Ah, I see, thanks for explaining